frontend/src/scenes/pipeline/hogfunctions/HogFunctionInputs.tsx

```diff
@@ -61,23 +62,41 @@ function JsonConfigField(props: {
     templating?: boolean
 }): JSX.Element {
     const { globalsWithInputs } = useValues(hogFunctionConfigurationLogic)
+    const key = useMemo(() => `json_field_${Math.random().toString(36).substring(2, 11)}`, [])
+
+    // Set up validation logic for this JSON field
+    const logic = hogFunctionInputLogic({
+        fieldKey: key,
+        initialValue: props.value,
+        onChange: props.onChange,
+    })
+
+    const { error } = useValues(logic)
+    const { setJsonValue } = useActions(logic)
+
+    // Format initial value for display
+    const formattedValue = useMemo(() => formatJsonValue(props.value), [props.value])
+
     return (
-        <CodeEditorResizeable
-            language={props.templating ? 'hogJson' : 'json'}
-            value={typeof props.value !== 'string' ? JSON.stringify(props.value, null, 2) : props.value}
-            onChange={(v) => props.onChange?.(v ?? '')}
-            options={{
-                lineNumbers: 'off',
-                minimap: {
-                    enabled: false,
-                },
-                scrollbar: {
-                    vertical: 'hidden',
-                    verticalScrollbarSize: 0,
-                },
-            }}
-            globals={props.templating ? globalsWithInputs : undefined}
-        />
+        <div className="flex flex-col gap-1 w-full">
+            <CodeEditorResizeable
+                language={props.templating ? 'hogJson' : 'json'}
+                value={formattedValue}
+                onChange={(value) => setJsonValue(value || '{}')}
+                options={{
+                    lineNumbers: 'off',
+                    minimap: {
+                        enabled: false,
+                    },
+                    scrollbar: {
+                        vertical: 'hidden',
+                        verticalScrollbarSize: 0,
+                    },
+                }}
+                globals={props.templating ? globalsWithInputs : undefined}
+            />
+            {error && <LemonField.Pure error={error} />}
```

Comment:
This fine but really the CodeEditorResizeable would be a child of the LemonField

Reply:
Storybook can be useful for learning how these things work https://storybook.posthog.net/?path=/docs/lemon-ui-lemon-field--docs